### PR TITLE
Add files for people

### PR DESCRIPTION
**Pull request by BartoszChrostowski**

### CODE REVIEW

File: backend/TreeOfAKind.API/PersonsFiles/PersonsFilesController.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        public async Task<IActionResult> AddPersonsFile([FromForm] AddPersonsFileRequest request)
+        {
+            var authId = HttpContext.GetFirebaseUserAuthId();
+
+            var file = request.File;
+
+            var result = await _mediator.Send(new AddPersonsFileCommand(authId, new TreeId(request.TreeId),
+                new Document(file.OpenReadStream(), file.ContentType, file.Name), new PersonId(request.PersonId)));
+
+            return Created(string.Empty, new IdUriDto{Id = result.Id.Value, Uri = result.FileUri});
```
Comment (Dragemil) [inline backend/TreeOfAKind.API/PersonsFiles/PersonsFilesController.cs:60]: TBH I don't know why would client need the file Uri right away

File: backend/TreeOfAKind.API/Trees/TreeController.cs [hunk 2]
```diff
@@ -225,7 +221,7 @@ public async Task<IActionResult> AddOrChangeTreePhoto([FromForm] AddTreePhotoReq
 
             var image = request.Image;
             var uri = await _mediator.Send(new AddOrChangeTreePhotoCommand(authId, new TreeId(request.TreeId),
-                new Document(image.OpenReadStream(), image.ContentType)));
+                new Document(image.OpenReadStream(), image.ContentType, "asdf")));
```
Comment (Dragemil) [inline backend/TreeOfAKind.API/Trees/TreeController.cs:224]: `asdf`

File: backend/TreeOfAKind.Application/Command/Document.cs [hunk 3]
```diff
@@ -6,11 +6,13 @@ public class Document
...
         public Stream Content { get; }
         public string ContentType { get; }
+        public string Name { get; }
 
-        public Document(Stream content, string contentType)
+        public Document(Stream content, string contentType, string name)
         {
             this.Content = content;
             ContentType = contentType;
+            Name = name;
```
Comment (Dragemil) [inline backend/TreeOfAKind.Application/Command/Document.cs:15]: Yeah, you have that here, but not in the API

File: backend/TreeOfAKind.Domain/Trees/People/NameOrLastNameMustBeSpecifiedRule.cs [hunk 4]
```diff
@@ -1,5 +1,4 @@
-﻿using System.Globalization;
-using TreeOfAKind.Domain.SeedWork;
+﻿using TreeOfAKind.Domain.SeedWork;
```
Comment (Dragemil) [inline backend/TreeOfAKind.Domain/Trees/People/NameOrLastNameMustBeSpecifiedRule.cs:1]: Still `Seedwork`

File: backend/TreeOfAKind.Domain/Trees/People/Person.cs [hunk 5]
```diff
@@ -65,5 +106,24 @@ private Person()
...
+        public File AddFile(string name, string mimeType, Uri fileUri)
+        {
+            var file = new File(name, mimeType, fileUri, this);
+            _files.Add(file);
+            return file;
+        }
+
+        public void RemoveFile(FileId fileId)
+        {
+            _files.RemoveAll(f => f.Id == fileId);
```
Comment (Dragemil) [inline backend/TreeOfAKind.Domain/Trees/People/Person.cs:119]: Event that will clear storage is missing?

File: backend/TreeOfAKind.Domain/Trees/Tree.cs [hunk 6]
```diff
@@ -122,5 +124,28 @@ public void RemoveTreePhoto()
...
+            var person = _people.First(p => p.Id == personId);
+            var file = person.AddOrChangeMainPhoto(name, mimeType, fileUri);
+            return file;
+        }
+
+        public void RemovePersonsFile(PersonId personId, FileId fileId)
+        {
+            CheckRule(new TreeMustContainPersonRule(People, personId));
+            var person = _people.First(p => p.Id == personId);
+            person.RemoveFile(fileId);
```
Comment (Dragemil) [inline backend/TreeOfAKind.Domain/Trees/Tree.cs:148]: Oh, that event should be here

File: backend/TreeOfAKind.Infrastructure/Domain/Trees/TreeEntityTypeConfiguration.cs [hunk 7]
```diff
@@ -59,6 +57,18 @@ public void Configure(EntityTypeBuilder<Tree> builder)
...
+
+                b.OwnsOne(p => p.MainPhoto, b =>
+                {
+                    b.ToTable("PersonsMainPhotos", SchemaNames.Trees);
+                    FileBuildAction(b);
+                });
+
+                b.OwnsMany(p => p.Files, b =>
+                {
+                    b.ToTable("PersonsMainFiles", SchemaNames.Trees);
```
Comment (Dragemil) [inline backend/TreeOfAKind.Infrastructure/Domain/Trees/TreeEntityTypeConfiguration.cs:69]: Will there be secondary files?

File: backend/Tests/TreeOfAKind.IntegrationTests/Files.cs [hunk 8]
```diff
@@ -24,7 +24,7 @@ public Files(ApplicationFixture applicationFixture) : base(applicationFixture)
 
         private async Task<Uri> AddExampleFile(TreeId treeId)
         {
-            var file = new Document(Stream.Null, "image/png");
+            var file = new Document(Stream.Null, "image/png", "ASDF.jpg");
```
[the remark this answers is not in the document]
Comment (BartoszChrostowski) [reply backend/Tests/TreeOfAKind.IntegrationTests/Files.cs:27]: ?

File: backend/TreeOfAKind.API/Trees/TreeController.cs (hunk 2 above)
Comment (BartoszChrostowski) [reply backend/TreeOfAKind.API/Trees/TreeController.cs:224]: `asdf`

File: backend/TreeOfAKind.Application/Command/Document.cs (hunk 3 above)
Comment (BartoszChrostowski) [reply backend/TreeOfAKind.Application/Command/Document.cs:15]: Name is taken from file

File: backend/TreeOfAKind.API/PersonsFiles/PersonsFilesController.cs (hunk 1 above)
Comment (BartoszChrostowski) [reply backend/TreeOfAKind.API/PersonsFiles/PersonsFilesController.cs:60]: Hmm idk, maybe, I'll leave it, u can ignore parameters when parsing :<

File: backend/TreeOfAKind.API/PersonsFiles/AddPersonsFileRequest.cs [hunk 9]
```diff
@@ -0,0 +1,13 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Http;
+
+namespace TreeOfAKind.API.PersonsFiles
+{
+    public class AddPersonsFileRequest
```
[the remark this answers is not in the document]
Comment (BartoszChrostowski) [reply backend/TreeOfAKind.API/PersonsFiles/AddPersonsFileRequest.cs:7]: Fileame is taken from uploaded file

**Comment by Dragemil** [comment]
Conflicts must be resolved

**Review by Dragemil — APPROVED** [review]

**Pull request merged**